mailpoet/tests/integration/Newsletter/AutomatedLatestContentAPITest.php

```diff
@@ -0,0 +1,228 @@
+
+    foreach ($this->createdUsers as $user) {
+      wp_delete_user($user->ID);
+    }
+
+    $this->deleteAllPosts();
+  }
+
+  private function loginWithRole(string $role): void {
+    $username = uniqid("testUser");
+    $email = "$username@test.com";
+    $existingUser = $this->wp->getUserBy("email", $email);
+
+    if ($existingUser) {
+      wp_delete_user($existingUser->ID);
+    }
+
+    wp_insert_user( [
+      'user_login' => $username,
+      'user_email' => $email,
+      'user_pass' => '',
+    ]);
+    $user = $this->wp->getUserBy("email", $email);
+
+    $user->add_role($role);
+
+    $this->wp->wpSetCurrentUser($user->ID);
+    $this->createdUsers[] = $user;
+  }
+
+  private function deleteAllPosts() {
+    global $wpdb;
+    $wpdb->query("TRUNCATE TABLE $wpdb->posts");
+  }
+
+  public function testGetBulkTransformedPosts() {
+    $this->deleteAllPosts();
+    $publishedPostTitle = 'Published Post';
+    wp_insert_post([
+      'post_title' => 'Private Post',
+      'post_content' => 'contents',
+      'post_status' => 'private',
+    ]);
+    wp_insert_post([
+      'post_title' => $publishedPostTitle,
+      'post_content' => 'contents',
+      'post_status' => 'publish',
+    ]);
+
+    $singleBlockQuery = array_merge(self::$alcBlock, ['postStatus' => "any"]);
+    $result = $this->alcAPI->getBulkTransformedPosts([
+      "blocks" => [$singleBlockQuery],
+    ]);
+    expect($result->data)->count(1);
+
+    expect(array_flatten($result->data)[0]->post_title)->equals($publishedPostTitle);
+
+    $this->loginWithRole("editor");
+    $result = $this->alcAPI->getBulkTransformedPosts([
+      "blocks" => [$singleBlockQuery],
+    ]);
+    expect($result->data)->count(1);
+    expect(array_flatten($result->data)[0]->post_title)->equals($publishedPostTitle);
+
+    $this->loginWithRole("administrator");
+    $result = $this->alcAPI->getBulkTransformedPosts([
+      "blocks" => [$singleBlockQuery],
+    ]);
+    expect($result->data)->count(1);
+    expect(array_flatten($result->data)[0]->post_title)->equals($publishedPostTitle);
+  }
+
+  /**
+   * @param \WP_Post[] $posts
+   * @return string[]
+   */
+  private function getPostTitles($posts): array {
+    return array_map(function (\WP_Post $post) {
+      return $post->post_title;// phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+    }, $posts);
+  }
+
+  public function testGetPosts() {
+    $this->deleteAllPosts();
+    $privatePost = [
+      'post_title' => 'Private Post',
+      'post_content' => 'contents',
+      'post_status' => 'private',
+    ];
+    $publicPost = [
+      'post_title' => 'Published Post',
+      'post_content' => 'contents',
+      'post_status' => 'publish',
+    ];
+    $draftPost = [
+      'post_title' => 'Draft Post',
+      'post_content' => 'contents',
+      'post_status' => 'draft',
+    ];
+
+    array_map(function ($postarr){
+      wp_insert_post($postarr);
+    }, [$privatePost, $publicPost, $draftPost]);
```

Comment:
Running `$privatePost = wp_insert_post([...])` directly would make it more consistent with the other test and would allow to get rid of this `array_map`.

Reply:
1) In this test I need access to `post_title` of all 3 created posts, removing these variables would mean I'll need to create another 3 variables which doesn't make any difference .
2) I'm not really sure what value  `consistency between different tests` can bring to us, nor do I know how getting rid of `array_map` can serve us

Reply:
1. Right, that explains it.
2. Not sure about the need for such a comment, I'm only trying to simplify code.